site/public/js/gradeable-notebook.js

```diff
@@ -79,13 +79,49 @@ function notebookAutosaveKey() {
  * Saves the current state of the notebook gradeable to localstorage.
  */
 function saveNotebookToLocal() {
-    if (typeof autosaveEnabled !== 'undefined' && autosaveEnabled) {
-        localStorage.setItem(notebookAutosaveKey(), JSON.stringify({
-            timestamp: Date.now(),
-            multiple_choice: gatherInputAnswersByType('multiple_choice'),
-            codebox: gatherInputAnswersByType('codebox'),
-        }));
-    }
+    const mc_inputs=[];
+    const short_answer_inputs = [];
+
+    //loop through multiple choice questions and save answers
+    $('.multiple_choice').each(function(){
+        let file_name='';
+        $(this).children('fieldset').each(function(){
+            file_name = $(this).data('filename');
+            if (file_name) {
+                const answers = [];
+                $(this).children('label').each(function(){
+                    //grab selected answers
+                    $(this).children('input').each(function(){
+                        if ($(this)[0].checked) {
+                            answers.push($(this)[0].defaultValue);
+                        }
+                    });
+                });
+                mc_inputs.push([file_name, answers]);
+            }
+        });
+    });
+
+    //save short answers
+    $('.short_answer').each(function(){
+        $(this).children('div').each(function(){
```

Comment:
Same as earlier, if someone were to add another level of div nesting outside of the answer, then this code stops working. Plus, targeting the `div` element is risky because it is so common, someone is bound to eventually mess up the structure. Instead, I would recommend this:
```suggestion
        $(this).find('div[name^="codebox_"]').each(function(){
```
This targets all divs that are children of the short answer (at any level), that also have a name attribute that starts with `codebox_`. This way our selector is more specific about the divs it wants so there is less chance of it pulling in garbage and the code becomes clearer.